test/proxy.js

```diff
@@ -117,6 +120,49 @@ test('connect through proxy (with pool)', async (t) => {
   pool.close()
 })
 
+test('connect through proxy to an https server', async (t) => {
+  t.plan(3)
+
+  const server = await buildHttpsServer()
+  const proxy = await buildProxy()
+
+  const serverUrl = `https://localhost:${server.address().port}`
+  const proxyUrl = `http://localhost:${proxy.address().port}`
+
+  server.on('request', (req, res) => {
+    t.equal(req.url, '/hello?foo=bar')
+    res.setHeader('content-type', 'application/json')
+    res.end(JSON.stringify({ hello: 'world' }))
+  })
+
+  const ca = readFileSync(join(__dirname, 'fixtures', 'ca.pem'), 'utf8')
+  const client = new Client(proxyUrl, {
+    tls: {
+      ca,
+      rejectUnauthorized: false,
+      maxCachedSessions: 1,
+      servername: 'agent1'
+    }
```

Comment:
I think you need to do the upgrade to tls yourself also please use `client.connect` instead of `client.request`.